frontend/src/components/smart/VariantChoiceModal.vue

```diff
@@ -0,0 +1,266 @@
+}
+
+export interface Variant {
+  id: number;
+  external_id: string;
+  name: string;
+  synced: boolean;
+  variant_id: number;
+  warehouse_product_variant_id?: number;
+  retail_price: string;
+  sku: string;
+  currency: string;
+  product: {
+    variant_id: number;
+    product_id: number;
+    image: string;
+    name: string;
+  };
+  files: File[];
+}
+
+interface File {
+  id: number;
+  type: FileType;
+  hash: string;
+  filename: string;
+  mime_type: string;
+  size: number;
+  width: number;
+  height: number;
+  dpi: number;
+  status: string;
+  created: number;
+  thumbnail_url: string;
+  preview_url: string;
+  visible: boolean;
+}
+
+export enum FileType {
+  DEFAULT = 'default',
+  PREVIEW = 'preview',
+}
+
+interface StoreMappedActions {
+  currentSkillPrice(): Promise<string>;
+}
+
+interface Data {
+  variants: Variant[];
+  selectedVariant?: Variant;
+  product?: Product;
+  quantity: number;
+  retailPriceInSkill: number;
+  totalPrice: number;
+  totalPriceInSkill: number;
+  skillPrice: number;
+}
+
+export default Vue.extend({
+  components: {CurrencyConverter},
+  data() {
+    return {
+      variants: [],
+      selectedVariant: undefined,
+      product: undefined,
+      quantity: 1,
+      retailPriceInSkill: 0,
+      totalPrice: 0,
+      totalPriceInSkill: 0,
+      skillPrice: 0,
+    } as Data;
+  },
+
+  computed: {
+    isMinusButtonDisabled() {
+      return this.$data.quantity <= 1;
+    },
+  },
+
+  methods: {
+    ...mapActions(['currentSkillPrice']) as StoreMappedActions,
+    fromWeiEther,
+    async fetchVariants(productId: number) {
+      const response = await api.getMerchandiseProductVariants(productId);
+      if (response.code !== 200) {
+        return;
+      }
+      this.variants = response.result.sync_variants;
+    },
+    addToCart() {
+      if (!this.product || !this.selectedVariant) return;
+
+      const cartEntry = {
+        product: this.product,
+        variant: this.selectedVariant,
+        quantity: this.quantity,
+      } as CartEntry;
+
+      this.$root.$emit('add-to-cart', cartEntry);
+      this.quantity = 1;
+    },
+    isFileTypePreview(file: File) {
+      return file.type === FileType.PREVIEW;
+    },
+    selectVariant(variant: Variant) {
+      this.selectedVariant = variant;
+      this.calculateTotalPrice();
+    },
+    subtractQuantity() {
+      this.quantity--;
+      this.calculateTotalPrice();
+    },
+    addQuantity() {
+      this.quantity++;
+      this.calculateTotalPrice();
+    },
+    calculateTotalPrice() {
+      if (!this.selectedVariant) return;
+      this.totalPrice = +this.selectedVariant.retail_price * this.quantity;
+      this.calculateTotalSkillPrice();
+    },
+    calculateTotalSkillPrice() {
+      if (!this.selectedVariant) return;
+      this.retailPriceInSkill = +this.selectedVariant.retail_price * this.skillPrice;
+      this.totalPriceInSkill = +this.selectedVariant.retail_price * this.quantity * this.skillPrice;
```

Comment:
Minor thing, but might as well use `this.totalPrice` that is already calculated here instead of `+this.selectedVariant.retail_price * this.quantity`.